### PR TITLE
ref(ts) Convert project action creators to typescript

### PR DESCRIPTION
**Pull request by markstory**

### CODE REVIEW

File: src/sentry/static/sentry/app/actionCreators/projects.tsx [hunk 1]
```diff
@@ -48,13 +58,13 @@ export function loadStats(api, params) {
 
 // This is going to queue up a list of project ids we need to fetch stats for
 // Will be cleared when debounced function fires
-const _projectStatsToFetch = new Set();
+const _projectStatsToFetch: Set<string> = new Set();
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/actionCreators/projects.tsx:61]: Could also just do `const _projectStatsToFetch = new Set<string>();

File: src/sentry/static/sentry/app/actionCreators/projects.tsx [hunk 2]
```diff
@@ -66,51 +76,53 @@ const _queryForStats = (api, projects, orgId) => {
...
-  _projectStatsToFetch.clear();
-}, 50);
+    // Split projects into more manageable chunks to query, otherwise we can
+    // potentially face server timeouts
+    const queries = chunk(projects, MAX_PROJECTS_TO_FETCH).map(chunkedProjects =>
+      _queryForStats(api, chunkedProjects, params.orgId)
+    );
 
-export function loadStatsForProject(api, project, params) {
+    Promise.all(queries)
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/actionCreators/projects.tsx:98]: Could convert to `await if you wanted`

```ts
try {
   const results = await Promise.all(queries);
   ProjectActions.loadStatsForProjectSuccess(
     results.reduce((acc, result) => acc.concat(result), [])
   );
} catch {
  addErrorMessage(t('Unable to fetch all project stats'));
}

// Reset projects list
_projectStatsToFetch.clear();
```

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry/static/sentry/app/actionCreators/projects.tsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/actionCreators/projects.tsx:61]: How would it know the contents of the set?

File: src/sentry/static/sentry/app/actionCreators/projects.tsx (hunk 2 above)
Comment (markstory) [reply src/sentry/static/sentry/app/actionCreators/projects.tsx:98]: I'll make another pass to add async/await.

**Pull request merged**